msal/src/main/java/com/microsoft/identity/nativeauth/statemachine/states/SignInStates.kt

```diff
@@ -382,55 +382,54 @@ class SignInPasswordRequiredState(
 
 /**
  * Native Auth uses a state machine to denote state of and transitions within a flow.
- * SignInAfterSignUpBaseState class is an abstract class to represent signin state after
- * successfull signup
- * in the signin flow.
+ * SignInContinuationState is a class to represent a sign in state after
+ * a successful signup or password reset.
  * @property continuationToken: Continuation token from signup APIS
  * @property username: Username of the user
  * @property config Configuration used by Native Auth
  */
-abstract class SignInAfterSignUpBaseState(
+class SignInContinuationState(
     override val continuationToken: String?,
-    internal open val username: String,
+    internal val username: String,
     private val config: NativeAuthPublicClientApplicationConfiguration
 ) : BaseState(continuationToken), State, Serializable {
-    private val TAG: String = SignInAfterSignUpBaseState::class.java.simpleName
+    private val TAG: String = SignInContinuationState::class.java.simpleName
 
     /**
-     * SignInAfterSignUpCallback receives the result for sign in after signup for Native Auth
+     * SignInContinuationCallback receives the result for sign in after flow completion for Native Auth
      */
-    interface SignInAfterSignUpCallback : Callback<SignInResult>
+    interface SignInContinuationCallback : Callback<SignInResult>
 
     /**
-     * Submits the sign-in-after-sign-up verification code to the server; callback variant.
+     * Submits the sign-in-continuation verification code to the server; callback variant.
      *
      * @param scopes (Optional) The scopes to request.
-     * @param callback [com.microsoft.identity.nativeauth.statemachine.states.SignInAfterSignUpBaseState.SignInAfterSignUpCallback] to receive the result on.
-     * @return The results of the sign-in-after-sign-up action.
+     * @param callback [com.microsoft.identity.nativeauth.statemachine.states.SignInContinuationState.SignInContinuationCallback] to receive the result on.
+     * @return The results of the sign-in-continuation action.
      */
-    fun signInAfterSignUp(scopes: List<String>? = null, callback: SignInAfterSignUpCallback) {
-        LogSession.logMethodCall(TAG, "${TAG}.signInAfterSignUp")
+    fun signIn(scopes: List<String>? = null, callback: SignInContinuationCallback) {
```

Comment:
Do we still need scopes in the case of a continuation token signIn?

Reply:
I think so - I can see scenarios where developers would want to request specific scopes when signing in the user after signing up. Otherwise they would have to start the sign in flow from scratch to request specific scopes.